src/sentry/rules/conditions/event_frequency.py

```diff
@@ -125,3 +127,75 @@ def query_hook(self, event, start, end, environment_id):
             environment_id=environment_id,
             use_cache=True,
         )[event.group_id]
+
+
+session_percent_intervals = {
+    "1m": ("one minute", timedelta(minutes=1)),
+    "5m": ("five minutes", timedelta(minutes=5)),
+    "10m": ("ten minutes", timedelta(minutes=10)),
+    "30m": ("30 minutes", timedelta(minutes=30)),
+    "1h": ("one hour", timedelta(minutes=60)),
+}
+
+
+class SessionPercentForm(EventFrequencyForm):
+    intervals = session_percent_intervals
+    interval = forms.ChoiceField(
+        choices=[
+            (key, label)
+            for key, (label, duration) in sorted(
+                session_percent_intervals.items(),
+                key=lambda key____label__duration: key____label__duration[1][1],
+            )
+        ]
+    )
+    value = forms.IntegerField(widget=forms.TextInput())
+
+
+class SessionPercentCondition(BaseEventFrequencyCondition):
+    label = "The issue has more errors than {value} percent of sessions in {interval}"
+
+    def __init__(self, *args, **kwargs):
+        self.intervals = session_percent_intervals
+        self.form_cls = SessionPercentForm
+        super().__init__(*args, **kwargs)
+
+    def query_hook(self, event, start, end, environment_id):
+        cache_key = f"r.c.spc:{environment_id}-{start.replace(minute=0, second=0, microsecond=0, tzinfo=None)}-{end.replace(minute=0, second=0, microsecond=0, tzinfo=None)}".replace(
+            " ", ""
+        )
+        session_count = cache.get(cache_key)
+        if session_count is None:
+            filters = {"project_id": [event.project_id]}
+            if environment_id:
+                filters["environment"] = [environment_id]
+
+            start = start - timedelta(days=5)
+            end = end + timedelta(days=5)
+            result_totals = raw_query(
+                selected_columns=["sessions"],
+                dataset=Dataset.Sessions,
+                start=start,
+                end=end,
+                filter_keys=filters,
+                groupby=["project_id"],
+                referrer="rules.conditions.event_frequency.SessionPercentCondition",
+            )
+            if result_totals["data"]:
+                session_count = result_totals["data"][0]["sessions"]
```

Comment:
Should we be summing all the values here per project?

Reply:
Nah, I don't think so because we are filtering by a single project. I was doing some testing recently and noticed that this will have to be summed up over time if/when we're querying for ranges greater than an hour (for now, we can keep the rollup at 60 and everything is in one bucket, but if rollup becomes smaller than the time range of the query, we see results like:
```[{'project_id': 2, 'bucketed_started': '2021-05-31T16:37:00+00:00', 'sessions': 2}, {'project_id': 2, 'bucketed_started': '2021-05-31T16:35:00+00:00', 'sessions': 2}]```.